File: backend/src/main/java/com/antshorttv/review/ReviewWorkbenchController.java
package com.antshorttv.review;

import com.antshorttv.common.ApiResponse;
import com.antshorttv.common.TenantRequestSupport;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.validation.Valid;
import java.util.List;
import org.springframework.core.io.Resource;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

@RestController
@RequestMapping("/api/script-review")
public class ReviewWorkbenchController {

    private final ReviewWorkbenchService reviewWorkbenchService;

    public ReviewWorkbenchController(ReviewWorkbenchService reviewWorkbenchService) {
        this.reviewWorkbenchService = reviewWorkbenchService;
    }

    @GetMapping("/projects")
    public ApiResponse<List<ReviewProjectSummaryResponse>> projects(HttpServletRequest request) {
        return ApiResponse.success(reviewWorkbenchService.listProjects(tenantId(request)));
    }

    @PostMapping(value = "/projects", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
    public ApiResponse<ReviewProjectDetailResponse> importProject(
        @RequestPart(value = "mainProjectId", required = false) Long mainProjectId,
        @RequestPart(value = "file", required = false) MultipartFile file,
        @RequestPart(value = "content", required = false) String content,
        @RequestPart(value = "name", required = false) String name,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.importProject(tenantId(request), mainProjectId, name, file, content));
    }

    @PutMapping("/projects/{reviewProjectId}/binding")
    public ApiResponse<ReviewProjectDetailResponse> bindProject(
        @PathVariable Long reviewProjectId,
        @Valid @RequestBody BindReviewProjectRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.bindProject(tenantId(request), reviewProjectId, body));
    }

    @GetMapping("/projects/{projectId}")
    public ApiResponse<ReviewProjectDetailResponse> project(
        @PathVariable Long projectId,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.detailProject(tenantId(request), projectId));
    }

    @PutMapping("/projects/{projectId}/versions")
    public ApiResponse<ReviewVersionResponse> saveVersion(
        @PathVariable Long projectId,
        @Valid @RequestBody SaveReviewVersionRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.saveVersion(tenantId(request), projectId, body));
    }

    @GetMapping("/projects/{projectId}/versions/{versionId}/history")
    public ApiResponse<ReviewVersionHistoryResponse> versionHistory(
        @PathVariable Long projectId,
        @PathVariable Long versionId,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.versionHistory(tenantId(request), projectId, versionId));
    }

    @PostMapping("/projects/{projectId}/tasks")
    public ApiResponse<ReviewTaskResponse> createTask(
        @PathVariable Long projectId,
        @Valid @RequestBody CreateReviewTaskRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.createTask(tenantId(request), projectId, body));
    }

    @GetMapping("/projects/{projectId}/tasks")
    public ApiResponse<List<ReviewTaskResponse>> tasks(
        @PathVariable Long projectId,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.listTasks(tenantId(request), projectId));
    }

    @GetMapping("/tasks/{taskId}")
    public ApiResponse<ReviewTaskResponse> task(
        @PathVariable Long taskId,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.taskDetail(tenantId(request), taskId));
    }

    @PutMapping("/tasks/{taskId}/config")
    public ApiResponse<ReviewTaskResponse> updateTaskConfig(
        @PathVariable Long taskId,
        @Valid @RequestBody UpdateReviewTaskRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.updateTaskConfig(tenantId(request), taskId, body));
    }

    @PostMapping("/tasks/{taskId}/cancel")
    public ApiResponse<ReviewTaskResponse> cancelTask(
        @PathVariable Long taskId,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.cancelTask(tenantId(request), taskId));
    }

    @PostMapping("/tasks/{taskId}/retry")
    public ApiResponse<ReviewTaskResponse> retryTask(
        @PathVariable Long taskId,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.retryTask(tenantId(request), taskId));
    }

    @PostMapping("/tasks/{taskId}/batch-repair")
    public ApiResponse<ReviewTaskResponse> batchRepair(
        @PathVariable Long taskId,
        @Valid @RequestBody BatchRepairReviewRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.batchRepair(tenantId(request), taskId, body));
    }

    @PostMapping("/issues/{issueId}/resolve")
    public ApiResponse<ReviewIssueResponse> resolveIssue(
        @PathVariable Long issueId,
        @Valid @RequestBody(required = false) MarkReviewIssueResolvedRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.markIssueResolved(tenantId(request), issueId, body));
    }

    @PostMapping("/projects/{projectId}/rollback")
    public ApiResponse<ReviewVersionResponse> rollback(
        @PathVariable Long projectId,
        @Valid @RequestBody RollbackReviewVersionRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.rollbackVersion(tenantId(request), projectId, body));
    }

    @PostMapping("/projects/{projectId}/exports")
    public ApiResponse<ReviewExportRecordResponse> export(
        @PathVariable Long projectId,
        @Valid @RequestBody ExportReviewReportRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(reviewWorkbenchService.exportReport(tenantId(request), projectId, body));
    }

    @GetMapping("/exports/{fileName}")
    public ResponseEntity<Resource> downloadExport(
        @PathVariable String fileName,
        HttpServletRequest request
    ) {
        Resource resource = reviewWorkbenchService.downloadExport(tenantId(request), fileName);
        return ResponseEntity.ok()
            .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"" + fileName + "\"")
            .contentType(MediaType.APPLICATION_OCTET_STREAM)
            .body(resource);
    }

    private Long tenantId(HttpServletRequest request) {
        return TenantRequestSupport.tenantId(request);
    }
}

File: frontend/src/services/ant-design-pro/reviewWorkbenchController.ts
// @ts-ignore
/* eslint-disable */
import { request } from "@umijs/max";

/** 此处后端没有提供注释 GET /api/script-review/exports/${param0} */
export async function downloadExport(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.downloadExportParams,
  options?: { [key: string]: any }
) {
  const { fileName: param0, ...queryParams } = params;
  return request<string>(`/api/script-review/exports/${param0}`, {
    method: "GET",
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /api/script-review/issues/${param0}/resolve */
export async function resolveIssue(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.resolveIssueParams,
  body: API.MarkReviewIssueResolvedRequest,
  options?: { [key: string]: any }
) {
  const { issueId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewIssueResponse>(
    `/api/script-review/issues/${param0}/resolve`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 GET /api/script-review/projects */
export async function projects(options?: { [key: string]: any }) {
  return request<API.ApiResponseListReviewProjectSummaryResponse>(
    "/api/script-review/projects",
    {
      method: "GET",
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/script-review/projects */
export async function importProject(
  body: {
    mainProjectId?: number;
    content?: string;
    name?: string;
  },
  file?: File,
  options?: { [key: string]: any }
) {
  const formData = new FormData();

  if (file) {
    formData.append("file", file);
  }

  Object.keys(body).forEach((ele) => {
    const item = (body as any)[ele];

    if (item !== undefined && item !== null) {
      if (typeof item === "object" && !(item instanceof File)) {
        if (item instanceof Array) {
          item.forEach((f) => formData.append(ele, f || ""));
        } else {
          formData.append(
            ele,
            new Blob([JSON.stringify(item)], { type: "application/json" })
          );
        }
      } else {
        formData.append(ele, item);
      }
    }
  });

  return request<API.ApiResponseReviewProjectDetailResponse>(
    "/api/script-review/projects",
    {
      method: "POST",
      data: formData,
      requestType: "form",
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 GET /api/script-review/projects/${param0} */
export async function project(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.projectParams,
  options?: { [key: string]: any }
) {
  const { projectId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewProjectDetailResponse>(
    `/api/script-review/projects/${param0}`,
    {
      method: "GET",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 PUT /api/script-review/projects/${param0}/binding */
export async function bindProject(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.bindProjectParams,
  body: API.BindReviewProjectRequest,
  options?: { [key: string]: any }
) {
  const { reviewProjectId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewProjectDetailResponse>(
    `/api/script-review/projects/${param0}/binding`,
    {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/script-review/projects/${param0}/exports */
export async function exportUsingPost(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.exportUsingPOSTParams,
  body: API.ExportReviewReportRequest,
  options?: { [key: string]: any }
) {
  const { projectId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewExportRecordResponse>(
    `/api/script-review/projects/${param0}/exports`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/script-review/projects/${param0}/rollback */
export async function rollback(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.rollbackParams,
  body: API.RollbackReviewVersionRequest,
  options?: { [key: string]: any }
) {
  const { projectId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewVersionResponse>(
    `/api/script-review/projects/${param0}/rollback`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 GET /api/script-review/projects/${param0}/tasks */
export async function tasks(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.tasksParams,
  options?: { [key: string]: any }
) {
  const { projectId: param0, ...queryParams } = params;
  return request<API.ApiResponseListReviewTaskResponse>(
    `/api/script-review/projects/${param0}/tasks`,
    {
      method: "GET",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/script-review/projects/${param0}/tasks */
export async function createTask(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.createTaskParams,
  body: API.CreateReviewTaskRequest,
  options?: { [key: string]: any }
) {
  const { projectId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewTaskResponse>(
    `/api/script-review/projects/${param0}/tasks`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 PUT /api/script-review/projects/${param0}/versions */
export async function saveVersion(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.saveVersionParams,
  body: API.SaveReviewVersionRequest,
  options?: { [key: string]: any }
) {
  const { projectId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewVersionResponse>(
    `/api/script-review/projects/${param0}/versions`,
    {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 GET /api/script-review/projects/${param0}/versions/${param1}/history */
export async function versionHistory(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.versionHistoryParams,
  options?: { [key: string]: any }
) {
  const { projectId: param0, versionId: param1, ...queryParams } = params;
  return request<API.ApiResponseReviewVersionHistoryResponse>(
    `/api/script-review/projects/${param0}/versions/${param1}/history`,
    {
      method: "GET",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 GET /api/script-review/tasks/${param0} */
export async function task(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.taskParams,
  options?: { [key: string]: any }
) {
  const { taskId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewTaskResponse>(
    `/api/script-review/tasks/${param0}`,
    {
      method: "GET",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/script-review/tasks/${param0}/batch-repair */
export async function batchRepair(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.batchRepairParams,
  body: API.BatchRepairReviewRequest,
  options?: { [key: string]: any }
) {
  const { taskId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewTaskResponse>(
    `/api/script-review/tasks/${param0}/batch-repair`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/script-review/tasks/${param0}/cancel */
export async function cancelTask(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.cancelTaskParams,
  options?: { [key: string]: any }
) {
  const { taskId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewTaskResponse>(
    `/api/script-review/tasks/${param0}/cancel`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 PUT /api/script-review/tasks/${param0}/config */
export async function updateTaskConfig(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.updateTaskConfigParams,
  body: API.UpdateReviewTaskRequest,
  options?: { [key: string]: any }
) {
  const { taskId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewTaskResponse>(
    `/api/script-review/tasks/${param0}/config`,
    {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/script-review/tasks/${param0}/retry */
export async function retryTask(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.retryTaskParams,
  options?: { [key: string]: any }
) {
  const { taskId: param0, ...queryParams } = params;
  return request<API.ApiResponseReviewTaskResponse>(
    `/api/script-review/tasks/${param0}/retry`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}
